src/urllib3/exceptions.py

```diff
@@ -285,7 +285,10 @@ class ProxySchemeUnknown(AssertionError, URLSchemeUnknown):
     # TODO(t-8ch): Stop inheriting from AssertionError in v2.0.
 
     def __init__(self, scheme):
-        message = f"Not supported proxy scheme {scheme}"
+        if scheme is None or scheme == "localhost":
```

Comment:
idea: add a branch like this so if we sometime fix our URL parser we know we can remove this hacky work-around. Also a comment about why `localhost` is being compared to a scheme might be warranted

```python
# 'localhost' is here because our URL parser parses localhost:8080 -> scheme=localhost, remove if we fix this
if scheme == "localhost":
    scheme = None
if scheme is None:
    message = ...
```

Reply:
Thanks @sethmlarson for pointing. I changed it to your suggested code block.